src/array_api_stubs/_draft/info.py

```diff
@@ -0,0 +1,149 @@
+__all__ = [
+    "__array_namespace_info__",
+    "capabilities",
+    "default_device",
+    "default_dtypes",
+    "devices",
+    "dtypes",
+]
+
+from ._types import (
+    Optional,
+    Union,
+    Tuple,
+    List,
+    device,
+    dtype,
+    DefaultDataTypes,
+    DataTypes,
+    Capabilities,
+    Info,
+)
+
+
+def __array_namespace_info__() -> Info:
+    """
+    Returns a namespace with Array API namespace inspection utilities.
+
+    Returns
+    -------
+    out: Info
+        Namespace with Array API namespace inspection utilities.
+    """
+
+
+def capabilities() -> Capabilities:
+    """
+    Returns a dictionary of array library capabilities.
+
+    The dictionary should contain the following keys:
+
+    -   **boolean_indexing**: boolean indicating whether an array library supports boolean indexing.
+    -   **data_dependent_shapes**: boolean indicating whether an array library supports data-dependent output shapes.
+
+    Returns
+    -------
+    out: Capabilities
+        a dictionary of array library capabilities.
+    """
+
+
+def default_device() -> device:
+    """
+    Returns the default device.
+
+    Returns
+    -------
+    out: device
+        an object corresponding to the default device.
+    """
+
+
+def default_dtypes(
+    *,
+    device: Optional[device] = None,
+) -> DefaultDataTypes:
+    """
+    Returns a dictionary containing default data types.
+
+    The dictionary should have the following keys:
+
+    -   **real floating**: default real floating-point data type.
+    -   **complex floating**: default complex floating-point data type.
```

Comment:
Q: I noticed the keys here (and later) have space between words, while those in `capabilities()` have underscores, maybe we can unify the style?

Reply:
These match the accepted "kind" values in the `isdtype` API. In hindsight, it may have been better to have the "kind" values be snakecase, as arguably more idiomatic, but that ship has sailed. In which case, I'll update the capabilities to use spaces.